### ISSUE TITLE
WinRatioByColor fails when only one color played

### ISSUE THREAD

**Issue by guidopetri** [guidopetri/chess-pipeline#65]
It's unclear whether this happens with 0 or 1 games being manipulated:

```
  File "chess-pipeline/src/newsletter.py", line 73, in run
    100 * color_stats.iloc[1][0],
IndexError: single positional indexer is out-of-bounds
```

Original reference is in #60 .

Fix steps are:

- [x] write a test to catch this/prevent this error
- [x] fix the error

**Comment by guidopetri** [comment]
Correction: this error comes up if there's only one color being played (i.e. only played games as black/white). Since Lichess tries to make you play both sides equally this usually only happens with one game.

**Comment by guidopetri** [comment]
This should be closed with 6f004a2 . I extracted the text generation out of `WinRatioByColor` and added some tests for it. Eventually I might want to do the same with `EloByWeekday`, but there aren't any cases where it breaks right now (afaik) so I'll leave it as part of the class for now.
